File: statuslogger.py
from colorama import init, Fore, Style

class StatusLogger:

    @staticmethod
    def progress(message: str):
        print(f"{Fore.BLUE}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def title(message: str):
        print(f"{Fore.YELLOW}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def warning(message: str):
        print(f"{Fore.LIGHTGREEN_EX}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def error(message: str):
        print(f"{Fore.RED}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def debug(message: str):
        print(f"{Fore.CYAN}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def success(message: str):
        print(f"{Fore.GREEN}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def info(message: str):
        print(f"{Fore.LIGHTWHITE_EX}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def notice(message: str):
        print(f"{Fore.LIGHTYELLOW_EX}", message, f"{Style.RESET_ALL}")

    @staticmethod
    def message(message: str):
        print(f"{Fore.WHITE}", message, f"{Style.RESET_ALL}")

File: statuslogger.mjs
/**
 * statuslogger.mjs — colorized status output (Node port of statuslogger.py).
 *
 * Uses raw ANSI escape codes; Node enables VT processing on Windows 10+ consoles automatically,
 * so no colorama-style shim is needed. Each method maps to the same level as the Python logger.
 */

const RESET = "\x1b[0m";
const COLOR = {
  blue: "\x1b[34m",
  yellow: "\x1b[33m",
  brightGreen: "\x1b[92m",
  red: "\x1b[31m",
  cyan: "\x1b[36m",
  green: "\x1b[32m",
  brightWhite: "\x1b[97m",
  brightYellow: "\x1b[93m",
  white: "\x1b[37m",
};

const emit = (color, message) => console.log(`${color}${message}${RESET}`);

export const StatusLogger = {
  progress: (message) => emit(COLOR.blue, message),
  title: (message) => emit(COLOR.yellow, message),
  warning: (message) => emit(COLOR.brightGreen, message),
  error: (message) => emit(COLOR.red, message),
  debug: (message) => emit(COLOR.cyan, message),
  success: (message) => emit(COLOR.green, message),
  info: (message) => emit(COLOR.brightWhite, message),
  notice: (message) => emit(COLOR.brightYellow, message),
  message: (message) => emit(COLOR.white, message),
};
